azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	props := resp.PropertiesType
+	if props != nil {
+		accounts := flattenVideoAnalyzerStorageAccounts(props.StorageAccounts)
+		if err := d.Set("storage_account", accounts); err != nil {
+			return fmt.Errorf("flattening `storage_account`: %s", err)
+		}
+	}
+
+	flattenedIdentity, err := flattenAzureRmVideoServiceIdentity(resp.Identity)
+	if err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
+	}
+
+	if err := d.Set("identity", flattenedIdentity); err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceVideoAnalyzerDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VideoAnalyzerID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("issuing AzureRM delete request for Video Analyzer '%s': %+v", id.Name, err)
+	}
+
+	return nil
+}
+
+func expandVideoAnalyzerStorageAccounts(d *pluginsdk.ResourceData) *[]videoanalyzer.StorageAccount {
+	storageAccountRaw := d.Get("storage_account").([]interface{})[0].(map[string]interface{})
+
+	results := []videoanalyzer.StorageAccount{
+		{
+			ID: utils.String(storageAccountRaw["id"].(string)),
+			Identity: &videoanalyzer.ResourceIdentity{
+				UserAssignedIdentity: utils.String(storageAccountRaw["identity_id"].(string)),
+			},
+		},
+	}
+
+	return &results
+}
+
+func flattenVideoAnalyzerStorageAccounts(input *[]videoanalyzer.StorageAccount) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, storageAccount := range *input {
+		output := make(map[string]interface{})
+
+		if storageAccount.ID != nil {
+			output["id"] = *storageAccount.ID
+		}
+
+		if storageAccount.Identity != nil {
+			output["identity_id"] = *storageAccount.Identity.UserAssignedIdentity
+		}
+
+		results = append(results, output)
+	}
+
+	return results
+}
+
+func expandAzureRmVideoAnalyzerIdentity(d *pluginsdk.ResourceData) (*videoanalyzer.Identity, error) {
+	identityRaw := d.Get("identity").([]interface{})
+	if identityRaw[0] == nil {
+		return nil, nil
+	}
+	identity := identityRaw[0].(map[string]interface{})
+	result := &videoanalyzer.Identity{
+		Type: utils.String(identity["type"].(string)),
+	}
+	var identityIdSet []interface{}
+	if identityIds, exists := identity["identity_ids"]; exists {
+		identityIdSet = identityIds.(*pluginsdk.Set).List()
+	}
+
+	// If type contains `UserAssigned`, `identity_ids` must be specified and have at least 1 element
+	if *result.Type == "UserAssigned" {
```

Comment:
there is technically a potential panic here, but since both fields are required in the schema this is fine

Reply:
I'll removed this if along with the below error, what do you think?

Reply:
That should be fine - if the schema changes we can add new validation accordingly 👍